airflow-core/src/airflow/ui/src/queries/useLogs.tsx

```diff
@@ -95,40 +94,72 @@ const parseLogs = ({
     return { data, warning };
   }
 
-  // TODO: Add support for nested groups
-
-  parsedLines = parsedLines.map((line) => {
-    const text = innerText(line);
-
-    if (text.includes("::group::") && !startGroup) {
-      startGroup = true;
-      groupName = text.split("::group::")[1] as string;
-    } else if (text.includes("::endgroup::")) {
-      startGroup = false;
-      const group = (
-        <details key={groupName} open={open} style={{ width: "100%" }}>
-          <summary data-testid={`summary-${groupName}`}>
-            <chakra.span color="fg.info" cursor="pointer">
-              {groupName}
-            </chakra.span>
-          </summary>
-          {groupLines}
-        </details>
-      );
-
-      groupLines = [];
-
-      return group;
-    }
+  parsedLines = (() => {
+    type Group = { level: number; lines: Array<JSX.Element | "">; name: string };
+    const groupStack: Array<Group> = [];
+    const result: Array<JSX.Element | ""> = [];
+
+    parsedLines.forEach((line) => {
+      const text = innerText(line);
+
+      if (text.includes("::group::")) {
+        const groupName = text.split("::group::")[1] as string;
+
+        groupStack.push({ level: groupStack.length, lines: [], name: groupName });
+
+        return;
+      }
 
-    if (startGroup) {
-      groupLines.push(line);
+      if (text.includes("::endgroup::")) {
+        const finishedGroup = groupStack.pop();
+
+        if (finishedGroup) {
+          const groupElement = (
+            <Box key={finishedGroup.name} mb={2} pl={finishedGroup.level * 2}>
+              <details open={open} style={{ width: "100%" }}>
```

Comment:
```suggestion
              <chakra.details open={open} w="100%">
```

to use [chakra factory](https://chakra-ui.com/docs/styling/chakra-factory) could preserve consistency and get better support for RTL lang